src/nspawn/nspawn.c

```diff
@@ -352,6 +352,15 @@ static int detect_unified_cgroup_hierarchy(void) {
         return 0;
 }
 
+static void parse_share_ns_env(const char *name, unsigned long mask) {
+        int r;
+
+        r = getenv_bool(name);
+        if (r < 0 && r != -ENXIO)
+                log_warning_errno(r, "Failed to parse %s from environment, defaulting to false.", name);
+        arg_share_ns = (arg_share_ns & ~mask) | (r > 0 ? 0 : mask);
```

Comment:
I am pretty sure if getenv_bool() returns -ENXIO we should not modify the mask at all. Right now you will remove the bit in that case.